precompiles/assets-erc20/src/eip2612.rs

```diff
@@ -264,7 +266,11 @@ where
 		asset_id: AssetIdOf<Runtime, Instance>,
 		handle: &mut impl PrecompileHandle,
 	) -> EvmResult<H256> {
-		handle.record_cost(RuntimeHelper::<Runtime>::db_read_gas_cost())?;
+		// Storage item: AssetMetadata:
+		// Blake2_128(16) + AssetId + AssetMetadata(16 + (2 * boundedString) + 2)
+		handle.record_db_read::<Runtime>(
+			34 + (2 * <Runtime as pallet_assets::Config<Instance>>::StringLimit::get()) as usize,
```

Comment:
We are missing at least 16 bytes here from AssetMetadata Balance.

> AssetMetadata(16 + (2 * boundedString) + 2)

Where are the 2 additional bytes coming from? encoding?